content/blog/kubernetes-monitoring-using-prometheus-and-grafana-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,419 @@
+---
+title: Adding a monitoring stack to a Kubernete cluster using Prometheus and
+  Grafana in HPE GreenLake for Private Cloud Enterprise
+date: 2024-01-25T14:13:04.918Z
+author: Guoping Jia
+authorimage: /img/guoping.png
+disable: false
+tags:
+  - Prometheus
+  - Grafana
+  - Kubernetes
+  - Kubernetes monitoring
+  - HPE GreenLake for Private Cloud Enterprise
+  - hpe-greenlake-for-private-cloud-enterprise
+---
+### Introduction
+
+[HPE GreenLake for Private Cloud Enterprise: Containers](https://www.hpe.com/us/en/greenlake/containers.html)("containers service"), one of the HPE GreenLake cloud services available on the HPE GreenLake for Private Cloud Enterprise, allows customers to create a Kubernetes (K8s) cluster, view details about existing clusters, and deploy containerized applications to the cluster. It provides an enterprise-grade container management service using open source K8s. 
+
+I﻿n this blog post, I will discuss K8s monitoring and show you how to add a monitoring stack using Prometheus and Grafana to a K8s cluster in HPE GreenLake for Private Cloud Enterprise. By setting up Prometheus as the data source and importing different dashboard templates into Grafana, various aspects of K8s, including metrics, performance, and health, can be monitored in the K8s cluster.
+
+### Why monitor K8s
+
+Though K8s dramatically simplifies application deployment in containers and across clouds, it adds a new set of complexities for managing, securing and troubleshooting applications. Container-based applications are dynamic and they are being designed using microservices, where the number of components is increased by an order of magnitude. To ensure K8s security, it requires self-configuration that is typically specified in code, whether K8s yaml manifests, Helm charts, or templating tools. Properly configuring for workloads, clusters, networks, and infrastructure is crucial for averting issues and limiting the impact if a breach occurs. Dynamic provisioning via Infrastructure as Code (IaC), automated configuration management and orchestration also add to monitoring and troubleshooting complexity. K8s monitoring is critical to managing application performance, service uptime and troubleshooting. Having a good monitoring tool is becoming essential for K8s monitoring. 
+
+### Prerequisites
+
+Before starting, make sure you meet the following requirements:
```

Comment:
Recommend rephrasing to:
"Before starting, make sure your setup has the following:"